src/node_os.cc

```diff
@@ -365,38 +366,31 @@ static void GetUserInfo(const FunctionCallbackInfo<Value>& args) {
                                                    pwd.username,
                                                    encoding,
                                                    &error);
+  if (username.IsEmpty()) {
+  fail_with_stringbytes_error:
+    uv_os_free_passwd(&pwd);
+    env->isolate()->ThrowException(error);
+    return;
+  }
```

Comment:
Suggestion: move the .IsEmpty() checks to after the StringBytes::Encode() calls, then you don't have to do the goto jump-around thing.